raiden/transfer/channel.py

```diff
@@ -1602,6 +1813,34 @@ def handle_refundtransfer(
     return is_valid, events, msg
 
 
+def handle_receive_lock_expired(
+        channel_state: NettingChannelState,
+        state_change: ReceiveLockExpired,
+):
+    """Remove expired locks from channel states."""
+    if state_change.transfer.channel_identifier != channel_state.identifier:
+        return TransitionResult(channel_state, [])
+
+    is_valid, msg, merkletree = is_valid_lockexpired(
+        state_change,
+        channel_state,
+        channel_state.partner_state,
+        channel_state.our_state,
+    )
+
+    events = list()
+    if is_valid:
+        delete_secrethash_endstate(channel_state.partner_state, state_change.secrethash)
```

Comment:
if the received balance proof is valid, the local state **must** be updated

Reply:
It is. The `delete_secrethash_endstate` function simply removes an entry from the `secrethashes_to_lockedlocks` dictionary. If this happens, the `TransitionResult` provides the current state of the channel after the entry was removed.